lib/chef/azure/commands/enable.rb

```diff
@@ -178,6 +179,23 @@ def configure_chef_only_once
     end
   end
 
+  def load_cloud_attributes_in_hints
+    cloud_attributes = {}
+    cloud_attributes["vm_name"] = Socket.gethostname
+    if windows?
+      vm_dns =  shell_out!("ipconfig").stdout
+      cloud_attributes["fqdn"] = vm_dns.gsub(/[a-zA-Z0-9-]*.[a-zA-Z0-9]*.[a-zA-Z0-9]*.cloudapp.net/).first.split('.')[0] + '.cloudapp.net' if vm_dns

```

Comment:
@adamedx  its public. No idea if there is something similar to this in Azure Ohai plugin.